File: Guarana.cpp
#include "CONSTANTS.h"
#include "Guarana.h"
#include "World.h"

Guarana::Guarana(World &world, const Point &position) : Plant(world, position, GUARANA_SYMBOL, GUARANA_STRENGTH, GUARANA_COLOR, GUARANA_NAME) {}

bool Guarana::Collision(Organism *attacker) {
  attacker->SetStrength(attacker->GetStrength() + 3);
  this->Die();
  this->world.messageManager.AddPlantEatingMessage(this->GetName(), attacker->GetName());
  return false;
}

Organism* Guarana::Clone(const Point &position) {
  return new Guarana(this->world, position);
}

File: Java/Guarana.java
public class Guarana extends Plant {
  public Guarana(World world, Point position) {
    super(world, position, Constants.GUARANA_SYMBOL, Constants.GUARANA_STRENGTH, Constants.GUARANA_COLOR, Constants.GUARANA_NAME);
  }

  @Override
  public final Organism Clone(Point position) {
    return new Guarana(this.world, position);
  }

  @Override
  public final boolean Collision(Organism other) {
    other.SetStrength(other.GetStrength() + 3);
    this.Die();
    this.world.messageManager.AddPlantEatingMessage(other.GetName(), this.GetName());
    return false;
  }
}
